test/OpenTelemetry.AutoInstrumentation.Tests/Util/ConfigurationSourceExtensionsTests.cs

```diff
@@ -0,0 +1,126 @@
+
+            list.Should().NotBeEmpty()
+                .And.HaveCount(3)
+                .And.ContainInOrder(new[] { TestEnum.Test1, TestEnum.Test2, TestEnum.Test3 })
+                .And.ContainItemsAssignableTo<TestEnum>();
+        }
+
+        [Fact]
+        public void ParseEnabledEnumList_Enabled()
+        {
+            var source = new NameValueConfigurationSource(new NameValueCollection()
+            {
+                { "TEST_ENABLED_VALUES", "Test1,Test3" }
+            });
+
+            var list = source.ParseEnabledEnumList<TestEnum>(
+                enabledConfiguration: "TEST_ENABLED_VALUES",
+                disabledConfiguration: "TEST_DISABLED_VALUES",
+                separator: ',',
+                error: "Invalid enum value: {0}");
+
+            list.Should().NotBeEmpty()
+                .And.HaveCount(2)
+                .And.ContainInOrder(new[] { TestEnum.Test1, TestEnum.Test3 })
+                .And.ContainItemsAssignableTo<TestEnum>();
+        }
+
+        [Fact]
+        public void ParseEnabledEnumList_Disabled()
+        {
+            var source = new NameValueConfigurationSource(new NameValueCollection()
+            {
+                { "TEST_DISABLED_VALUES", "Test2" }
+            });
+
+            var list = source.ParseEnabledEnumList<TestEnum>(
+                enabledConfiguration: "TEST_ENABLED_VALUES",
+                disabledConfiguration: "TEST_DISABLED_VALUES",
+                separator: ',',
+                error: "Invalid enum value: {0}");
+
+            list.Should().NotBeEmpty()
+                .And.HaveCount(2)
+                .And.ContainInOrder(new[] { TestEnum.Test1, TestEnum.Test3 })
+                .And.ContainItemsAssignableTo<TestEnum>();
+        }
+
+        [Fact]
+        public void ParseEnabledEnumList_None()
+        {
+            var source = new NameValueConfigurationSource(new NameValueCollection()
+            {
+                { "TEST_ENABLED_VALUES", "none" }
+            });
+
+            var list = source.ParseEnabledEnumList<TestEnum>(
+                enabledConfiguration: "TEST_ENABLED_VALUES",
+                disabledConfiguration: "TEST_DISABLED_VALUES",
+                separator: ',',
+                error: "Invalid enum value: {0}");
+
+            list.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void ParseEnabledEnumList_Invalid()
+        {
+            var source = new NameValueConfigurationSource(new NameValueCollection()
+            {
+                { "TEST_ENABLED_VALUES", "invalid" }
+            });
+
+            source.Invoking(s => s.ParseEnabledEnumList<TestEnum>(
+                enabledConfiguration: "TEST_ENABLED_VALUES",
+                disabledConfiguration: "TEST_DISABLED_VALUES",
+                separator: ',',
+                error: "Invalid enum value: {0}"))
+                .Should().Throw<FormatException>()
+                .WithMessage("Invalid enum value: invalid");
+        }
```

Comment:
Maybe this would be more readable?

```csharp
            var act = () => source.ParseEnabledEnumList<TestEnum>(
                enabledConfiguration: "TEST_ENABLED_VALUES",
                disabledConfiguration: "TEST_DISABLED_VALUES",
                separator: ',',
                error: "Invalid enum value: {0}");

            act.Should().Throw<FormatException>().WithMessage("Invalid enum value: invalid");
```

just check if it works as I write it from my memory 😉